### PR TITLE
Adding SAI api tests for CRUD operation

### PR DESCRIPTION
**Pull request by murthyvijay**
requesting review of SAI api tests

### CODE REVIEW

File: test/test-cases/scale/saic/sai-api/test_sai_api_vnet_acl_in.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
+import json
+import time
+from pathlib import Path
+from pprint import pprint
+
+import pytest
+import saichallenger.dataplane.snappi.snappi_traffic_utils as stu
```
Comment (chrispsommers) [inline test/test-cases/scale/saic/sai-api/test_sai_api_vnet_acl_in.py:7]: If these test cases are only to test APIs and no traffic, let's remove all unused imports. Make the files as lean as possible.

File: test/test-cases/scale/saic/sai-api/test_sai_api_vnet_acl_in.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+current_file_dir = Path(__file__).parent
+
+# Constants
+SWITCH_ID = 5
+
+
+class TestSaiVnetAclIn:
+
+    def test_vnet_acl_in_create(self, dpu):
```
Comment (chrispsommers) [inline test/test-cases/scale/saic/sai-api/test_sai_api_vnet_acl_in.py:24]: More specifically, this is an IPv4 ACL group create. There are multiple aspects to making a useful "ACL" - creating a group, creating rule(s) and adding those rules to the group. So you should decide the scope of each test case (just create a group, or create a group and rules and add to the group); also consider the permutations of IPv4 and IPv6, etc.
Here is a good reference of PTF test cases: https://github.com/Azure/DASH/blob/746d482e2a022356bd82084c3f3e463343d004ac/test/test-cases/functional/ptf/saidashacl.py#L83

File: test/test-cases/scale/saic/sai-api/test_sai_api_vnet_eni.py [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+                "type": "SAI_OBJECT_TYPE_ENI",
+                "atrribute": "SAI_ENI_ATTR_VM_UNDERLAY_DIP"
+
+            }
+        ]
+        result = [*dpu.process_commands(commands)]
+        print("\n======= SAI commands RETURN values get =======")
+        pprint(result)
+
+        assert (result[0].value(), "20.10.2.10")
```
Comment (chrispsommers) [inline test/test-cases/scale/saic/sai-api/test_sai_api_vnet_eni.py:195]: change to proper expression - replace` ,` with `==`

**Review by chrispsommers — CHANGES_REQUESTED** [review]
See comments - need to clean up unused code and fix the asserts. Great start, I like the approach.

**Comment by chrispsommers** [comment]
Another thought - since these are all "functional tests," should they go under a new directory `test/test-cases/functional/saic/sai-api`?

File: test/test-cases/scale/saic/sai-api/test_sai_api_vnet_acl_in.py [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+                "op": "remove",
+                "type": "SAI_OBJECT_TYPE_DASH_ACL_GROUP",
+                "attributes": [
+                    "SAI_DASH_ACL_GROUP_ATTR_IP_ADDR_FAMILY",
+                    "SAI_IP_ADDR_FAMILY_IPV4"
+                ]
+            },
+        ]
+
+        result = [*dpu.process_commands(commands)]
```
Comment (chrispsommers) [inline test/test-cases/scale/saic/sai-api/test_sai_api_vnet_acl_in.py:92]: no assert? test for return value == 0

File: test/test-cases/scale/saic/sai-api/test_sai_api_vnet_inbound_vpe.py [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+        print("\n======= SAI commands RETURN values get =======")
+        pprint(result)
+
+        assert (result[0].value() == "SAI_VIP_ENTRY_ACTION_REJECT")
+
+    def test_vnet_inbound_simple_remove(self, dpu):
+
+        commands = [
+            {
+                "name": "vpe",
```
Comment (chrispsommers) [inline test/test-cases/scale/saic/sai-api/test_sai_api_vnet_inbound_vpe.py:93]: should this be `$vpe` ?

File: test/test-cases/scale/saic/sai-api/test_sai_api_vnet_acl_in.py [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+        print("\n======= SAI commands RETURN values get =======")
+        pprint(result)
+
+        assert (result[0].value() == "SAI_IP_ADDR_FAMILY_IPV4")
+
+    def test_vnet_acl_in_remove(self, dpu):
+
+        commands = [
+            {
+                "name": "acl_in",
```
Comment (chrispsommers) [inline test/test-cases/scale/saic/sai-api/test_sai_api_vnet_acl_in.py:82]: should this be `$acl_in` ?

**Review by chrispsommers — CHANGES_REQUESTED** [review]
See comments

File: test/test-cases/scale/saic/sai-api/test_sai_api_vnet_acl_in.py (hunk 6 above)
Comment (murthyvijay) [reply test/test-cases/scale/saic/sai-api/test_sai_api_vnet_acl_in.py:82]: No, the other scripts worked fine with just the name and no variable.

File: test/test-cases/scale/saic/sai-api/test_sai_api_vnet_inbound_vpe.py (hunk 5 above)
Comment (murthyvijay) [reply test/test-cases/scale/saic/sai-api/test_sai_api_vnet_inbound_vpe.py:93]: No, the other scripts worked fine with just the name and no variable.

**Pull request merged**